x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java

```diff
@@ -626,9 +647,18 @@ private void assertInternalConsistency() {
             || (getAuthenticatingSubject().getMetadata().get(AuthenticationField.API_KEY_ID_KEY) != null)
             : "API KEY authentication requires metadata to contain API KEY id, and the value must be non-null.";
 
+        assert (false == isAuthenticatedAsRemoteAccess())
+            || (getAuthenticatingSubject().getMetadata().get(AuthenticationField.API_KEY_ID_KEY) != null)
```

Comment:
I opted for re-using the API key metadata fields as is (i.e., `REMOTE_ACCESS` authentication has an `API_KEY_ID_KEY` instead of a new `REMOTE_ACCESS_API_KEY_ID_KEY` field. This makes for a smaller change, and will make it easier to build roles in next PR (we can re-use the API key role descriptors parsing logic, for the API key specific role descriptor fields, for instance).